compiler/rustc_interface/src/passes.rs

```diff
@@ -883,6 +905,8 @@ fn run_required_analyses(tcx: TyCtxt<'_>) {
                 CStore::from_tcx(tcx).report_unused_deps(tcx);
             },
             {
+                tcx.ensure_ok().exportable_items(LOCAL_CRATE);
+                tcx.ensure_ok().stable_order_of_exportable_impls(LOCAL_CRATE);
```

Comment:
This logic shouldn't run if we are not building a sdylib (or its interface?).

Also, since sdylibs aren't supposed to have metadata, these queries' results don't need to be encoded into metadata, and they don't need extern providers.

Reply:
> Also, since sdylibs aren't supposed to have metadata, these queries' results don't need to be encoded into metadata, and they don't need extern providers.

They need to be encoded. Interface metadata is used to get symbol for calling an "exportable" instance.